src/codegen/operator/index_scan_translator.cpp

```diff
@@ -0,0 +1,343 @@
+    for (unsigned int i = 0; i < where_clause_attributes.size(); i++) {
+      const auto *ai = where_clause_attributes[i];
+      llvm::Value *attribute_id = codegen.Const32(ai->attribute_id);
+      llvm::Value *attribute_name = codegen.ConstStringPtr(ai->name);
+      bool is_lower_key = false;
+      if (comparison_type[i] == peloton::ExpressionType::COMPARE_GREATERTHAN ||
+          comparison_type[i] ==
+              peloton::ExpressionType::COMPARE_GREATERTHANOREQUALTO) {
+        is_lower_key = true;
+      }
+      llvm::Value *is_lower = codegen.ConstBool(is_lower_key);
+
+      llvm::Value *parameter_value = parameter_cache.GetValue(i).GetValue();
+      switch (ai->type.type_id) {
+        case peloton::type::TypeId::TINYINT:
+        case peloton::type::TypeId::SMALLINT:
+        case peloton::type::TypeId::INTEGER: {
+          codegen.Call(IndexScanIteratorProxy::UpdateTupleWithInteger,
+                       {iterator_ptr, parameter_value, attribute_id,
+                        attribute_name, is_lower});
+          break;
+        }
+        case peloton::type::TypeId::TIMESTAMP:
+        case peloton::type::TypeId::BIGINT: {
+          codegen.Call(IndexScanIteratorProxy::UpdateTupleWithBigInteger,
+                       {iterator_ptr, codegen->CreateSExt(parameter_value,
+                                                          codegen.Int64Type()),
+                        attribute_id, attribute_name, is_lower});
+          break;
+        }
+        case peloton::type::TypeId::DECIMAL: {
+          if (parameter_value->getType() != codegen.DoubleType()) {
+            codegen.Call(
+                IndexScanIteratorProxy::UpdateTupleWithDouble,
+                {iterator_ptr,
+                 codegen->CreateSIToFP(parameter_value, codegen.DoubleType()),
+                 attribute_id, attribute_name, is_lower});
+          } else {
+            codegen.Call(IndexScanIteratorProxy::UpdateTupleWithDouble,
+                         {iterator_ptr, parameter_value, attribute_id,
+                          attribute_name, is_lower});
+          }
+          break;
+        }
+        case peloton::type::TypeId::VARBINARY:
+        case peloton::type::TypeId::VARCHAR: {
+          codegen.Call(IndexScanIteratorProxy::UpdateTupleWithVarchar,
+                       {iterator_ptr, parameter_value, attribute_id,
+                        attribute_name, is_lower});
+          break;
+        }
+        case peloton::type::TypeId::BOOLEAN: {
+          codegen.Call(IndexScanIteratorProxy::UpdateTupleWithBoolean,
+                       {iterator_ptr, parameter_value, attribute_id,
+                        attribute_name, is_lower});
+          break;
+        }
+        default: {
+          throw new Exception("Type" +
+                              peloton::TypeIdToString(ai->type.type_id) +
+                              " is not supported in codegen yet");
+        }
+      }
+    }
+  }
+}
+}
```

Comment:
Formatting - add a new line between 341 and 342, then add `// namespace codegen` and `// namespace peloton` on the last two lines.